client/src/components/admin/tables/ProjectRow.tsx

```diff
@@ -67,19 +74,32 @@ const ProjectRow = ({ project, idx, checked, handleCheckedChange }: ProjectRowPr
         }
     };
 
+    useEffect(() => {
+        // Filter flags to find all matching ones
+        const matchingFlags = flags.filter((flag) => flag.project_id === project.id);
+        // Set matching flag count
+        const matchingFlagCount = matchingFlags.length;
+        // If there is at least one matching flag, set the first one as the matching flag
+        if (matchingFlagCount > 0) {
+            setMatchingFlag(matchingFlags[0]); // Update matchingFlag state to the first match
+            setMatchingFlagCount(matchingFlagCount); // Update matchingFlagCount state
+            setProjectIDState(matchingFlags[0].project_id); // Update projectIDState based on first matching flag
+        } else {
+            setMatchingFlag(undefined); // No match found
+            setProjectIDState(''); // Clear project ID state
+        }
+    }, [flags, project]); // Recalculate when flags or project changes
+
+    const rowClassNameFlag = [
+        'border-t-2 border-backgroundDark duration-150',
+        matchingFlag ? 'bg-error bg-opacity-30' : '',
+        checked ? 'bg-primary/20' : !project.active ? 'bg-lightest' : 'bg-background',
+    ]
+        .filter(Boolean)
```

Comment:
why do we need to filter here? aren't all items in this list gonna be a string?

Reply:
also, I have [tailwind-merge](https://www.npmjs.com/package/tailwind-merge) installed, so you could just do this inline in the JSX